src/essreflectometry/amor/types.py

```diff
@@ -0,0 +1,54 @@
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+from ..reflectometry.types import Run
+
+# TODO What do they mean?
+# Supermirror parameters
+MValue = NewType('MValue', str)
+CriticalEdge = NewType('CriticalEdge', str)
+Alpha = NewType('Alpha', str)
+SupermirrorParameter = TypeVar('SupermirrorParameter', MValue, CriticalEdge, Alpha)
+
+
+class Supermirror(sciline.Scope[SupermirrorParameter, sc.Variable], sc.Variable):
+    """Supermirror parameter scope."""
```

Comment:
What are you using a generic provider here? If it is only for grouping, define things in a submodule, e.g., `supermirror.MValue` instead of `SupermirrorParameter[MValue]`.